dts/bindings/input/zephyr,input-lvgl-button.yaml

```diff
@@ -0,0 +1,47 @@
+# Copyright 2023 Fabian Blatz <fabianblatz@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  LVGL button indev pseudo-device
+
+  Listens for button input events and routes the
+  lv_indev_data_t to the underlying button lv_indev_t managed by LVGL.
+
+  Needs to be associated to a specific device to listen for events
+  from that device. Example configuration:
+
+  pointer {
+          compatible = "zephyr,input-lvgl-button";
+          input = <&buttons>;
+          input-codes = <INPUT_KEY_0>;
+          lvgl-key = <10>;
+          x-coordinate = <120>;
+          y-coordinate = <120>;
```

Comment:
Oh that's interesting, so the button can either map to an `lvgl-key` or a coordinate right? How about splitting this in two, `zephyr,input-lvgl-button-key` and `zephyr,input-lvgl-button-coordinate`, then you would have an array of codes mapping to either an array or keys or an array of coordinates. That way you could do all keys to whatever combination of coordinate and (lvgl) keys in at most two definitions, and also saves you from having to do build check based validation.

Reply:
From what I understood and also while testing it seems like it goes like this:
If the assigned `lv_indev_data_t->key` is `LV_KEY_ENTER (10)` then the press and release are relayed as a click at the coordinates of the button.
Otherwise there are keys with a special interaction: [https://docs.lvgl.io/latest/en/html/overview/indev.html](https://docs.lvgl.io/latest/en/html/overview/indev.html)